playground/Testcases.tsx

```diff
@@ -0,0 +1,114 @@
+import { useRef, useEffect, Fragment } from "react";
+
+import { FLOWCHART_DIAGRAM_TESTCASES } from "./testcases/flowchart";
+import { SEQUENCE_DIAGRAM_TESTCASES } from "./testcases/sequence.ts";
+import { CLASS_DIAGRAM_TESTCASES } from "./testcases/class.ts";
+import { UNSUPPORTED_DIAGRAM_TESTCASES } from "./testcases/unsupported.ts";
+
+import type { UpdateMermaidDefinition } from "./index.tsx";
+import { MermaidDiagram } from "./MermaidDiagram.tsx";
+
+interface TestCaseProps {
+  name: string;
+  testcases: { name: string; definition: string }[];
+  onChange: (definition: string, activeTestcaseIndex: number) => void;
+  error: string | null;
+  activeTestcase?: number;
+}
+
+const Testcase = ({
+  name,
+  testcases,
+  onChange,
+  error,
+  activeTestcase,
+}: TestCaseProps) => {
+  const baseId = name.toLowerCase();
+
+  useEffect(() => {
+    if (activeTestcase !== undefined) {
+      const { definition } = testcases[activeTestcase];
+
+      onChange(definition, activeTestcase);
+    }
+  }, [testcases]);
+
+  return (
+    <>
+      <h2>
+        {name} {"Diagrams"}
+      </h2>
+      <details>
+        <summary>
+          {name} {"Examples"}
+        </summary>
+        <div id={`${baseId}-container`} className="testcase-container">
+          {testcases.map(({ name, definition }, index) => {
+            const id = `${baseId}-${index}`;
+            return (
+              <Fragment key={id}>
+                <h2 style={{ marginTop: "50px", color: "#f06595" }}>{name}</h2>
+                <pre>{definition}</pre>
+                <button
+                  onClick={() => {
+                    onChange(definition, index);
+                  }}
+                >
+                  {"Render to Excalidraw"}
+                </button>
+
+                <MermaidDiagram definition={definition} id={id} />
+
+                {error && activeTestcase === index && (
+                  <div id="error">{error}</div>
+                )}
+              </Fragment>
+            );
+          })}
+        </div>
+      </details>
+    </>
+  );
+};
+
+interface TestcasesProps {
+  onChange: UpdateMermaidDefinition;
+  isCustomTest: boolean;
+  error: TestCaseProps["error"];
+}
+
+const Testcases = ({ onChange, error, isCustomTest }: TestcasesProps) => {
+  const activeTestcase = useRef<[number, number]>();
+
+  const testCases = [
+    { name: "Flowchart", testcases: FLOWCHART_DIAGRAM_TESTCASES },
+    { name: "Sequence", testcases: SEQUENCE_DIAGRAM_TESTCASES },
+    { name: "Class", testcases: CLASS_DIAGRAM_TESTCASES },
+    { name: "Unsupported", testcases: UNSUPPORTED_DIAGRAM_TESTCASES },
+  ];
+
+  return (
+    <>
+      {testCases.map(({ name, testcases }, index) => (
+        <Testcase
+          key={index}
+          name={name}
+          activeTestcase={
+            activeTestcase.current?.[0] === index
+              ? activeTestcase.current[1]
+              : undefined
```

Comment:
this is confusing, lets have rename to `activeTestCaseIndex` and only store the active test case index here.
You can instead compare it with `index` here and pass the prop accordingly

Reply:
@ad1992 now I'm a bit confused haha. 
The idea was to store both the index of the test files and the index of the active testcase. Without this, it doesn't know that it's file X that has the active test, and it ends up requesting to render the Excalidraw for all files not only for current active testcase.

Reply:
@igorwessel to fix that can we have a base index instead? It starts from `0` and keeps on incrementing until all test cases are rendered.

Reply:
@ad1992 I will try this approach it should work. if don't I think it might be better to just destructure and name these variables
```js
const [activeTestCaseFileIndex, activeTestCaseId] = activeTestCaseIndex.current

activeTestCaseFileIndex === index ? activeTestCaseId : undefined
```

Reply:
@ad1992 I change the mind of  some of the responsibility and understood that the duty of triggering the active test should lie with TestCases, not with SingleTestCase
done in [378eaf9](https://github.com/excalidraw/mermaid-to-excalidraw/pull/52/commits/378eaf978d72b4f17a7382a9aadcfb6fb4e789ba)